src/meta/MetaHttpIngestHandler.cpp

```diff
@@ -0,0 +1,168 @@
+        err_ = HttpCode::E_ILLEGAL_ARGUMENT;
+        return;
+    }
+
+    path_ = headers->getQueryParam("path");
+    space_ = headers->getIntQueryParam("space");
+}
+
+void MetaHttpIngestHandler::onBody(std::unique_ptr<folly::IOBuf>) noexcept {
+    // Do nothing, we only support GET
+}
+
+
+void MetaHttpIngestHandler::onEOM() noexcept {
+    switch (err_) {
+        case HttpCode::E_UNSUPPORTED_METHOD:
+            ResponseBuilder(downstream_)
+                .status(WebServiceUtils::to(HttpStatusCode::METHOD_NOT_ALLOWED),
+                        WebServiceUtils::toString(HttpStatusCode::METHOD_NOT_ALLOWED))
+                .sendWithEOM();
+            return;
+        case HttpCode::E_ILLEGAL_ARGUMENT:
+            ResponseBuilder(downstream_)
+                .status(WebServiceUtils::to(HttpStatusCode::BAD_REQUEST),
+                        WebServiceUtils::toString(HttpStatusCode::BAD_REQUEST))
+                .sendWithEOM();
+            return;
+        default:
+            break;
+    }
+
+    if (ingestSSTFiles(space_, path_)) {
+        LOG(INFO) << "SSTFile ingest successfully " << path_;
+        ResponseBuilder(downstream_)
+            .status(WebServiceUtils::to(HttpStatusCode::OK),
+                    WebServiceUtils::toString(HttpStatusCode::OK))
+            .body("SSTFile ingest successfully")
+            .sendWithEOM();
+    } else {
+        LOG(ERROR) << "SSTFile ingest failed";
+        ResponseBuilder(downstream_)
+            .status(WebServiceUtils::to(HttpStatusCode::FORBIDDEN),
+                    WebServiceUtils::toString(HttpStatusCode::FORBIDDEN))
+            .body("SSTFile ingest failed")
+            .sendWithEOM();
+    }
+}
+
+void MetaHttpIngestHandler::onUpgrade(UpgradeProtocol) noexcept {
+    // Do nothing
+}
+
+
+void MetaHttpIngestHandler::requestComplete() noexcept {
+    delete this;
+}
+
+
+void MetaHttpIngestHandler::onError(ProxygenError error) noexcept {
+    LOG(ERROR) << "Web Service MetaHttpIngestHandler got error : "
+               << proxygen::getErrorString(error);
+}
+
+bool MetaHttpIngestHandler::ingestSSTFiles(GraphSpaceID space, const std::string& path) {
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto prefix = MetaServiceUtils::partPrefix(space);
+    auto ret = kvstore_->prefix(0, 0, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        LOG(ERROR) << "Fetch Parts Failed";
+        return false;
+    }
+
+    std::set<std::string> storageIPs;
+    while (iter->valid()) {
+        for (auto host : MetaServiceUtils::parsePartVal(iter->val())) {
+            auto storageIP = network::NetworkUtils::intToIPv4(host.get_ip());
+            if (std::find(storageIPs.begin(), storageIPs.end(), storageIP) == storageIPs.end()) {
+                storageIPs.insert(std::move(storageIP));
+            }
+        }
+        iter->next();
+    }
+
+    std::vector<folly::SemiFuture<bool>> futures;
+    static nebula::thread::GenericThreadPool pool;
```

Comment:
if we want to stop the meta service, how can we stop the pool?

Reply:
when running `GenericThreadPool`'s destructor , `stop()` will be called.